commands/utility/schedule.js

```diff
@@ -0,0 +1,118 @@
+const { SlashCommandBuilder, PermissionFlagsBits } = require('discord.js');
+
+/*  ----------------------
+Google Calendar API Setup
+-------------------------- */
+const { google } = require('googleapis');
+
+const auth = new google.auth.GoogleAuth({
+	keyFile: "./credentials.json",
+	scopes: "https://www.googleapis.com/auth/calendar.readonly"
+})
+const calendar = google.calendar({ version: "v3", auth });
+const calendarId = "c2e4ddc715d346cd957fadf163edcded4b957fcb6381d5e4b5f503c660689de5@group.calendar.google.com";
+/* ------------------- */
+
+
+module.exports = {
+	data: new SlashCommandBuilder()
+		.setName('schedule')
+    .setDescription('Display upcoming meetings and workshops!'),
+  async execute(interaction) {
+    // get user info
+    const user = interaction.guild.members.cache.get(interaction.member.id)
+
+    // get array of user's roles
+    let rolesList = await getRoles(user);
+
+    // get next 15 events from calendars
+    const res = await calendar.events.list({
+      calendarId: calendarId,
+      timeMin: new Date().toISOString(),
+      maxResults: 15,
+      singleEvents: true,
+      orderBy: 'startTime',
+    });
+    const allEvents = res.data.items;
+
+    let events = [];
+
+    // sort through events and only get add the ones that are applicable by role
```

Comment:
wait how are events determined? if the string of the role is in the calendar string?

Reply:
to be solved; sherwin thinking/discussing with management team(?)
currently determined by roles being in calendar event description


- can intake multiple roles (i.e. user must have both CF Skateboard and electrical roles to be in the CF Skateboard Electrical Meeting)
- have not set it up to handle different non-overlapping roles together, can do so if you want

Reply:
sg lets just go with your logic for now and if we needa change it we can